src/Engine/ProtoCore/Lang/CallSite.cs

```diff
@@ -174,11 +174,6 @@ public ISerializable GetLeftMostData()
 
                 if (!HasNestedData)
                     return null;
-#if DEBUG
-
-                Validity.Assert(NestedData != null, "Nested data has changed null status since last check, suspected race");
-                Validity.Assert(NestedData.Count > 0, "Empty subnested array, please file repo data with @lukechurch, relates to MAGN-4059");
```

Comment:
@reddyashish I think it makes sense to keep the `NestedData != null` assertion and just remove the second one as there is a check for `NestedData.Count` to be zero in the next line.